packages/sources/finnhub/src/endpoint/quote.ts

```diff
@@ -79,39 +78,52 @@ export const buildSymbol = ({
   return base
 }
 
-const requestTransform = (req: AdapterRequest<typeof inputParameters.validated>) => {
-  req.requestContext.data.base = req.requestContext.data.base.toUpperCase()
-  req.requestContext.data.quote = req.requestContext.data.quote?.toUpperCase()
-  req.requestContext.data.exchange = req.requestContext.data.exchange?.toUpperCase()
+const requestTransform =
+  (defaultExchange: string) => (req: AdapterRequest<typeof inputParameters.validated>) => {
+    req.requestContext.data.base = req.requestContext.data.base.toUpperCase()
+    req.requestContext.data.quote = req.requestContext.data.quote?.toUpperCase()
+    req.requestContext.data.exchange = req.requestContext.data.exchange?.toUpperCase()
 
-  const requestBase = req.requestContext.data.base
-  const requestExchange = req.requestContext.data.exchange
+    const requestBase = req.requestContext.data.base
+    const requestQuote = req.requestContext.data.quote
+    const requestExchange = req.requestContext.data.exchange
 
-  // If the symbol is a full symbol (containing Exchange and Pair), split into the separate params
-  if (isExchangeSymbol(requestBase)) {
-    const params = splitSymbol(requestBase)
+    // If the symbol is a full symbol (containing Exchange and Pair), split into the separate params
+    if (isExchangeSymbol(requestBase)) {
+      const params = splitSymbol(requestBase)
 
-    // If exchange is provided in request, prefer that over the exchange split from the symbol
-    const exchange = requestExchange || params.exchange
+      // If exchange is provided in request, prefer that over the exchange split from the symbol
+      const exchange = requestExchange || params.exchange
 
-    if (exchange && exchanges[exchange] == undefined) {
-      throw new AdapterInputError({
-        statusCode: 400,
-        message: `${exchange} is not a supported exchange, must be one of [${Object.keys(
-          exchanges,
-        ).join(', ')}]`,
-      })
+      req.requestContext.data.base = params.base
+      req.requestContext.data.quote = params.quote
+      req.requestContext.data.exchange = exchange
+
+      return
     }
 
```

Comment:
It would be good to have a debug log here showing the original request data in case we ever have any issues with unexpected values, eg. forgetting that the job spec has a specific exchange specified

Reply:
Good shout! I've added for both cases where we alter the incoming request